gempython/tools/amc_user_functions_xhal.py

```diff
@@ -279,7 +274,15 @@ def getGBTLinkStatus(self,doReset=False,printSummary=False, ohMask=None):
         gbtMonData = OHLinkMonitorArrayType()
         self.getmonGBTLink(gbtMonData, self.nOHs, ohMask, doReset)
 
-        NGBT = gbtsPerGemVariant["ge11"]
+        #Determine the number of GBTs based on the gemType
+        if gemType == "ge11":
+            NGBT = gbtsPerGemVariant["ge11"]
+        elif gemType == "ge21":
+            NGBT = gbtsPerGemVariant["ge21"]
+        else:
+            NGBT = 0
+            print("Unrecognized gemType, setting NGBT = 0")
+
```

Comment:
Would strongly prefer the following:

```python
if gemType in gbtsPerGemVariant.keys():
    self.NGBT = gbtsPerGemVariant[gemType]
else:
    raise KeyError("Unrecognized gemType {0}".format(gemType))
```

Reply:
An alternative, but also acceptable implementation in `HwAMC::_init__(...), could also be:

```python
try:
   self.NGBT = gbtsPerGemVariant[gemType]
except KeyError as error:
   printRed("Unrecognized gemType {0}".format(gemType))
   exit(os.EX_DATAERR)
```

You might need to add `import os` and `from gempython.utils.gemLogger import printRed`.  I believe `printRed` is defined under `gempython/utils/gemLogger.py` but double check.

Reply:
I agree, I will implement and resubmit the pull request.